templates/nextcloud28-entrypoint.sh

```diff
@@ -0,0 +1,125 @@
+#!/bin/bash
+
+
+chown -R www-data:www-data /var/www/html
+chmod -R 775 /var/www/html
+
+# Let /entrypoint.sh not run apache
+sed -i 's/exec "$@"//g' /entrypoint.sh
+
+# create file to override config
+cat > /var/www/html/config/liquid.override.config.php << 'DELIM'
+<?php
+$CONFIG = array (
+  'dbhost' =>  getenv('MYSQL_HOST'),
+  'trusted_domains' => [
+                    "localhost",
+                    getenv('NEXTCLOUD_HOST'),
+                    getenv('LIQUID_BASE_IP')
+],
+  'dbname' => getenv('MYSQL_DATABASE'),
+  'dbpassword' => getenv('MYSQL_PASSWORD'),
+);
+DELIM
+
+# Launch canonical entrypoint
+/entrypoint.sh apache2-foreground
+
+unzip /apps-to-install/sociallogin.zip -d /apps-to-install
+mkdir -p /var/www/html/custom_apps/sociallogin
+mv /apps-to-install/nextcloud-social-login-*/* /var/www/html/custom_apps/sociallogin
+rm -r /apps-to-install/nextcloud-social-login-*
+rm /apps-to-install/*.zip
+
+chown -R www-data:www-data /var/www/html
+chmod -R 775 /var/www/html
+
+# Function from the Nextcloud's original entrypoint
+run_as() {
+    if [ "$(id -u)" = 0 ]; then
+        su -p www-data -s /bin/bash -c "$1"
+    else
+        sh -c "$1"
+    fi
+}
+
+run_as 'php occ config:system:set dbhost --value $MYSQL_HOST'
+run_as 'php occ app:enable sociallogin'
+
+OAUTH_SETTINGS=$(cat << DELIM
+'{"custom_oauth2":
+[
+{"name":"$OAUTH2_PROVIDER_NAME",
+"title":"$OAUTH2_PROVIDER_TITLE",
+"apiBaseUrl":"$OAUTH2_BASE_URL",
+"authorizeUrl":"$OAUTH2_AUTHORIZE_URL",
+"tokenUrl":"$OAUTH2_TOKEN_URL",
+"profileUrl":"$OAUTH2_PROFILE_URL",
+"logoutUrl":"$OAUTH2_LOGOUT_URL",
+"clientId":"$OAUTH2_CLIENT_ID",
+"clientSecret":"$OAUTH2_CLIENT_SECRET",
+"scope":"read",
+"profileFields":"$OAUTH2_PROFILE_FIELDS",
+"displayNameClaim":"$OAUTH2_NAME_FIELD",
+"groupsClaim":"$OAUTH2_GROUPS_CLAIM",
+"style":"",
+"defaultGroup":"",
+"groupMapping":{"$OAUTH2_ADMIN_GROUP":"admin"}}
+]}'
+DELIM
+)
+
+run_as "php occ config:app:set sociallogin custom_providers --value=$OAUTH_SETTINGS"
+
+
+# run_as 'php occ config:app:set sociallogin auto_create_groups --value=1'
```

Comment:
explicitly set this to what you want it to be (1 or 0) to override any breakage users make by manually clicking on the admin button.